interface/requests/response.go

```diff
@@ -6,172 +6,187 @@ package requests
 import (
 	"bytes"
 	"fmt"
+	"reflect"
 
-	"github.com/sirupsen/logrus"
 	"google.golang.org/protobuf/proto"
 )
 
-const responseHeaderSizeValue uint16 = 30
-const responseHeaderSize uint16 = responseHeaderSizeValue + 6
+const wireHeaderSizeValue uint16 = 30
+const WireHeaderSize uint16 = wireHeaderSizeValue + 6
 
-// Response codes
+type StatusCode uint16
+
+// Statusonse codes
 const (
-	RespSuccess                      uint16 = 0
-	RespWrongProviderID              uint16 = 1
-	RespContentTypeNotSupported      uint16 = 2
-	RespAcceptTypeNotSupported       uint16 = 3
-	RespVersionTooBig                uint16 = 4
-	RespProviderNotRegistered        uint16 = 5
-	RespProviderDoesNotExist         uint16 = 6
-	RespDeserializingBodyFailed      uint16 = 7
-	RespSerializingBodyFailed        uint16 = 8
-	RespOpcodeDoesNotExist           uint16 = 9
-	RespResponseTooLarge             uint16 = 10
-	RespUnsupportedOperation         uint16 = 11
-	RespAuthenticationError          uint16 = 12
-	RespAuthenticatorDoesNotExist    uint16 = 13
-	RespAuthenticatorNotRegistered   uint16 = 14
-	RespKeyDoesNotExist              uint16 = 15
-	RespKeyAlreadyExists             uint16 = 16
-	RespPsaErrorGenericError         uint16 = 1132
-	RespPsaErrorNotPermitted         uint16 = 1133
-	RespPsaErrorNotSupported         uint16 = 1134
-	RespPsaErrorInvalidArgument      uint16 = 1135
-	RespPsaErrorInvalidHandle        uint16 = 1136
-	RespPsaErrorBadState             uint16 = 1137
-	RespPsaErrorBufferTooSmall       uint16 = 1138
-	RespPsaErrorAlreadyExists        uint16 = 1139
-	RespPsaErrorDoesNotExist         uint16 = 1140
-	RespPsaErrorInsufficientMemory   uint16 = 1141
-	RespPsaErrorInsufficientStorage  uint16 = 1142
-	RespPsaErrorInssuficientData     uint16 = 1143
-	RespPsaErrorCommunicationFailure uint16 = 1145
-	RespPsaErrorStorageFailure       uint16 = 1146
-	RespPsaErrorHardwareFailure      uint16 = 1147
-	RespPsaErrorInsufficientEntropy  uint16 = 1148
-	RespPsaErrorInvalidSignature     uint16 = 1149
-	RespPsaErrorInvalidPadding       uint16 = 1150
-	RespPsaErrorTamperingDetected    uint16 = 1151
+	StatusSuccess                      StatusCode = 0
+	StatusWrongProviderID              StatusCode = 1
+	StatusContentTypeNotSupported      StatusCode = 2
+	StatusAcceptTypeNotSupported       StatusCode = 3
+	StatusVersionTooBig                StatusCode = 4
```

Comment:
For this one the spec has: `WireProtocolVersionNotSupported` ([here](https://parallaxsecond.github.io/parsec-book/parsec_client/status_codes.html))

Reply:
updated